pulpcore/pulpcore/app/serializers/repository.py

```diff
@@ -435,5 +427,20 @@ class Meta:
         model = models.RepositoryVersion
         fields = ModelSerializer.Meta.fields + (
             '_href', '_content_href', '_added_href', '_removed_href', 'number',
-            'content_summary', 'add_content_units', 'remove_content_units', 'base_version'
+            'content_summary', 'base_version'
         )
+
+
+class RepositoryVersionCreateSerializer(ModelSerializer, NestedHyperlinkedModelSerializer):
+    add_content_units = serializers.ListField(
+        help_text=_('A list of content units to add to a new repository version'),
+        write_only=True
+    )
+    remove_content_units = serializers.ListField(
+        help_text=_('A list of content units to remove from the latest repository version'),
+        write_only=True
+    )
+
+    class Meta:
+        model = models.RepositoryVersion
+        fields = ['add_content_units', 'remove_content_units']
```

Comment:
So when creating, are these the only fields that will be serialized? Is that a usability problem?

Reply:
I don't understand the question. Please rephrase.

Reply:
Also how does moving these fields here resolve the `write_only` issue? I don't understand this unfortunately.

Reply:
The problem is that we had 1 serializer for both GET and POST. However, DRF, was able to distinguish which fields were to be used for accepting POST body params and which fields to use when serializing for a GET. DRF knows this because some fields are 'write_only'. However, the drf_yasg library we use for generating OpenAPI schema does not take into account the 'write_only' attribute of a field. As a result when it was using the serializer to generate a schema for the GET response it was including the 'write_only' fields.

Reply:
Does it make any sense to include the other fields also on this serializer?

Reply:
this serializer is only used to inspect data in the POST body. the response is the 202 OperationPosteponed